code/game/objects/structures/ghost_role_spawners.dm

```diff
@@ -106,15 +106,18 @@
 	travel the stars with a single declaration: \"Yeah go do whatever.\" Though you are bound to the one who created you, it is customary in your society to repeat those same words to newborn \
 	golems, so that no golem may ever be forced to serve again.</b>"
 
-/obj/effect/mob_spawn/human/golem/New()
+/obj/effect/mob_spawn/human/golem/New(loc, datum/species/golem/species, has_owner, mob/creator)
 	..()
+	if(species)
+		name += " ([initial(species.id)])"
+		mob_species = species
 	var/area/A = get_area(src)
 	if(A)
-		notify_ghosts("A golem shell has been completed in \the [A.name].", source = src, action=NOTIFY_ATTACK)
-	spawn(1)//give it time to get an owner
-		if(owner)
-			flavour_text = "You are a golem. You move slowly, but are highly resistant to heat and cold as well as blunt trauma. You are unable to wear clothes, but can still use most tools. \
-			Serve [owner], and assist [owner.p_them()] in completing their goals at any cost."
+		notify_ghosts("\A [initial(species.id)] golem shell has been completed in \the [A.name].", source = src, action=NOTIFY_ATTACK)
+	if(has_owner && creator)
+		flavour_text = "You are a golem. You move slowly, but are highly resistant to heat and cold as well as blunt trauma. You are unable to wear clothes, but can still use most tools. \
```

Comment:
it's the same message as adamantines, i guess people'll manage to figure that out even for mineral golems